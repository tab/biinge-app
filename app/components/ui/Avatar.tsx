import React from "react"
import { View, ViewStyle, StyleProp } from "react-native"
import FastImage from "react-native-fast-image"
import { MD5 } from "crypto-js"

import { avatarStyles } from "styles"

type Props = {
  style?: StyleProp<ViewStyle>
  email: string
  size?: number
}

const AvatarComponent = ({ style, email, size }: Props) => {
  const uri = `https://www.gravatar.com/avatar/${MD5(
    email,
  ).toString()}/?s=${size}&d=mp`

  return (
    <View style={[avatarStyles.root, style]}>
      <FastImage
        style={avatarStyles.image}
        source={{
          uri: uri,
          priority: FastImage.priority.normal,
        }}
        resizeMode={FastImage.resizeMode.cover}
      />
    </View>
  )
}

AvatarComponent.defaultProps = {
  size: 150,
}

export default AvatarComponent
