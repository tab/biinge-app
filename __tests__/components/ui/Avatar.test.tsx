import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"

import { it } from "@jest/globals"

import Avatar from "components/ui/Avatar"

describe("Avatar", () => {
  const email = "test@local"

  it("renders correctly", () => {
    const tree = render(<Avatar email={email} />)
    expect(tree).toMatchSnapshot()
  })
})
