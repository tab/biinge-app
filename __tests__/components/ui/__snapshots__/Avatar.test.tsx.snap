// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 50,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "priority": "normal",
          "uri": "https://www.gravatar.com/avatar/00a95fd18d173c5542ca052cc7ad9638/?s=150&d=mp",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
